syntax = "proto3";

import "enums.proto";
import "assets.proto";

message ArtistId {
    int32 user_id = 1;
}

message Artist {
    int32 user_id = 1;
    string user_name = 2;
    Location location = 3;
    int32 range = 4; //Distance away from location they are willing to travel (in metres)
    Price price = 5;
    ContactInfo contact_info = 6;
    Rating rating = 7;
    Genre genre = 8;
    int32 booking_count = 9;
}

service Artists {
    rpc GetArtistName(ArtistId) returns (Artist);
}